### ISSUE TITLE
Question about AdversarialDebiasing.py

### ISSUE THREAD

**Issue by AnakokEmre** [giandos200/Zhang-et-al-Mitigating-Unwanted-Biases-with-Adversarial-Learning#1]
Hello,
On the **line 278** of the code for [AdversarialDebiasing.py]( https://github.com/giandos200/Zhang-et-al-Mitigating-Unwanted-Biases-with-Adversarial-Learning/blob/main/model/Torch/AdversarialDebiasing.py) you have written 
adv_grad = [torch.clone(par.grad.detach()) for par in **self.clf_model**.parameters()]
Shouldn't it be 
adv_grad = [torch.clone(par.grad.detach()) for par in **self.adv_model**.parameters()] 
instead ?

**Comment by giandos200** [comment]
Hi @AnakokEmre ,
Thank you for your question; if you see the paper and, in particular, equation 1, the adversary gradient and the projection is respect the parameter of the Neural Network and not the Adversary.
$\Delta_W LP - proj_{\Delta_W LA}(\Delta_W LP) - \alpha \Delta_W LA$ 

line 278 calculates the gradient of the weights of the neural network with respect to the opposing loss and then performs the projection with respect to the gradient of the loss of the neural network itself. In fact at line 263/264 the backward is made with respect to CLF and 274/275 the backward is made with respect to ADV. If you try to change with the self.adv_model.parameters you will have (i) mismatch of parameters and, sequentially, (ii) dimension problem. 

I hope everything is now clear!

**Comment by AnakokEmre** [comment]
Thank you very much for your answer !
